### PR TITLE
[FIX] registry: no name in constraint check

### PR DESCRIPTION
**Pull request by nseinlet**
When table name is long enough, the foreign key constraint name computed 
by odoo and the one by PostgreSQL are different, for the same table, 
column and definition. 

As an example, base_partner_merge_automatic_wizard_res_partner_rel(partner_id) foreign 
key is named base_partner_merge_automatic_wizard_res_partner_rel_res_partner_id_fkey 
in Odoo while it's base_partner_merge_automatic_wizard_res_par_res_partner_id_fkey in 
PostgreSQL. This difference trigger a useless foreign key drop/create.

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: odoo/modules/registry.py [hunk 1]
```diff
@@ -495,7 +495,7 @@ def check_foreign_keys(self, cr):
             if spec is None:
                 sql.add_foreign_key(cr, table1, column1, table2, column2, ondelete)
                 model.env['ir.model.constraint']._reflect_constraint(model, conname, 'f', None, module)
-            elif spec != (conname, table2, column2, deltype):
+            elif (spec[1], spec[2], spec[3]) != (table2, column2, deltype):
                 sql.drop_constraint(cr, table1, spec[0])
                 sql.add_foreign_key(cr, table1, column1, table2, column2, ondelete)
                 model.env['ir.model.constraint']._reflect_constraint(model, conname, 'f', None, module)
```
Comment (KangOl) [inline odoo/modules/registry.py:501]: Actually, the `conname` stored in `ir.model.constraint` will be wrong. 
Actually, I think that's the computation of `conname` that should be fixed instead.

**Review by KangOl — CHANGES_REQUESTED** [review]

File: odoo/modules/registry.py (hunk 1 above)
Comment (rco-odoo) [reply odoo/modules/registry.py:501]: I agree with @KangOl : `conname` is indeed the source of the problem.
I actually wonder whether it is safe to ignore the constraint name in this context...

**Review by rco-odoo — APPROVED** [review]
@robodoo r+

**Comment by rco-odoo** [comment]
@robodoo r+

**Pull request closed without merging**